npm/src/updateconsignmentstatus/index.ts

```diff
@@ -18,23 +19,28 @@ export class UpdateConsignmentStatus {
 
   async markConsignmentStatusAsCompleted(
     uploadFilesInfo: FileUploadInfo
-  ): Promise<number> {
-    const variables: MarkUploadAsCompletedMutationVariables = {
-      consignmentId: uploadFilesInfo.consignmentId
-    }
+  ): Promise<number | void> {
+    try {
+      const variables: MarkUploadAsCompletedMutationVariables = {
+        consignmentId: uploadFilesInfo.consignmentId
+      }
 
-    const result: FetchResult<MarkUploadAsCompletedMutation> =
-      await this.client.mutation(MarkUploadAsCompleted, variables)
+      const result: FetchResult<MarkUploadAsCompletedMutation> =
+        await this.client.mutation(MarkUploadAsCompleted, variables)
 
-    if (!result.data || !result.data.markUploadAsCompleted || result.errors) {
-      const errorMessage: string = result.errors
-        ? result.errors.toString()
-        : "no data"
-      throw Error(
-        `Marking the Consignment Status as "Completed" failed: ${errorMessage}`
+      if (!result.data || !result.data.markUploadAsCompleted || result.errors) {
+        const errorMessage: string = result.errors
+          ? result.errors.toString()
+          : "no data"
+        throw Error(errorMessage)
+      } else {
+        return result.data.markUploadAsCompleted
+      }
+    } catch (e) {
+      handleUploadError(
```

Comment:
This function is about calling the API, so making it call a method which changes the UI mixes up the levels of abstraction, which I think will make it harder to find this code. Where is the `markConsignmentStatusAsCompleted` method called, and does it deal with code at the page level? Would moving the `catch` and the call to `handleUploadError` fit better there?